test/exercises/wavelet/ecgIO.chpl

```diff
@@ -0,0 +1,25 @@
+use IO, IO.FormattedIO;
+import Subprocess.spawn;
+
+proc readEcgData(path: string): [] int throws {
+  iter getSamples(fr) {
+    var a, b: int;
+    while fr.readf("%i %i\n", a, b) {
+      yield b;
+    }
+  }
+  const samples = getSamples(openReader(path));
+  return samples;
```

Comment:
This could probably be simplified to `return getSamples(openReader(path));`?  Or...  why not just make `readEcgData()` the iterator rather than a procedure with a nested iterator (pretty advanced, likely to lead to "why would I do this?" questions), which I think would still work at the callsite?  This would avoid the complication of the nested routine while also permitting the same routine either to generate an array or in an iterator/loop context.